plugins/jobs/notify.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = schema.models.job.base.extract('id');
+const statusSchema = schema.models.build.base.extract('status');
+const messageSchema = joi.string();
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/jobs/{id}/notify',
+    options: {
+        description: 'Get a single pipeline',
+        notes: 'Returns a pipeline record',
+        tags: ['api', 'jobs'],
+        auth: {
+            strategies: ['token'],
+            scope: ['pipeline']
+        },
+
+        handler: async (request, h) => {
+            const { jobFactory, pipelineFactory } = request.server.app;
+            const { credentials } = req.auth;
+            const jobId = request.params.id;
+            const job = await jobFactory.get(jobId);
+
+            if (!job) {
+                throw boom.notFound(`Job ${id} does not exist`);
+            }
+
+            const pipelineId = job.pipelineId;
+
+            if (pipelineId !== credentials.pipelineId) {
+                throw boom.forbidden('Token does not have permission for this pipeline');
+            }
+
+            const pipeline = pipelineFactory.get(pipelineId);
+
+            await request.server.events.emit('job_status', {
```

Comment:
should this be wrapped in a try/catch with some error logging?

Reply:
i am just following the pattern from place where it emits build_status

Reply:
and also it's just emitting an event from server, i don't think this will be the point of failure.